frontend/src/components/Header.tsx

```diff
@@ -1,20 +1,96 @@
-import { NavLink } from "react-router-dom";
-import { NAV } from "../utils/consts";
+import { useState } from "react";
+import { NavLink, useNavigate } from "react-router-dom";
+import { AiOutlineClose, AiOutlineMenu } from "react-icons/ai";
+
 import Login from "./Login";
+import { NAV, NAV_ITEMS } from "../lib/consts";
+import { Button } from "./UI/button";
+import { useAppDispatch } from "../hooks/redux";
+import { toggleNav } from "../store/uiSlice";
 
 function Header() {
+  const [isNav, setIsNav] = useState(false);
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+
+  const handleNav = () => {
+    setIsNav(prev => !prev);
+    dispatch(toggleNav());
+  };
+
+  const handleNavigateHome = () => {
+    navigate("/");
+  };
+
   return (
-    <header className="w-full max-h-10 bg-teal-300">
-      <nav className="w-full flex justify-between gap-3 px-6">
-        <ul className="w-full flex justify-between">
-          {Object.keys(NAV).map(nav => (
-            <NavLink key={nav} to={NAV[nav] || "/"}>
-              {nav}
-            </NavLink>
+    <header className="bg-white flex justify-between items-center px-4 text-[#3a0e7b] fixed top-0 z-50 w-full">
+      <nav className="w-full flex justify-between gap-3 px-3 py-4 sm:px-6 sm:py-6">
+        {/* Desktop Navigation Items */}
+        <ul className="w-full hidden md:flex items-center justify-between">
+          <li>
+            <Button
```

Comment:
Make sure all interactive elements have appropriate aria-* attributes for accessibility

Reply:
we have created a separate task for that